src/tsf/Implementation.cpp

```diff
@@ -403,13 +413,89 @@ STDMETHODIMP Implementation::EditSessionProxyBase::QueryInterface(REFIID riid, v
 
 ULONG STDMETHODCALLTYPE Implementation::EditSessionProxyBase::AddRef() noexcept
 {
-    return InterlockedIncrement(&referenceCount);
+    InterlockedIncrement(&referenceCount);
+    return self->AddRef();
 }
 
 ULONG STDMETHODCALLTYPE Implementation::EditSessionProxyBase::Release() noexcept
 {
-    FAIL_FAST_IF(referenceCount == 0);
-    return InterlockedDecrement(&referenceCount);
+    InterlockedDecrement(&referenceCount);
+    return self->Release();

```

Comment:
Mind explaining this? I can't fully grok it.

Reply:
This ensures that `self` (aka parent aka the Implementation instance) stays alive for the duration of the request.